client/components/Timer.tsx

```diff
@@ -0,0 +1,98 @@
+'use client';
+import { useState, useEffect, useRef } from 'react';
+import { Button } from './ui/button';
+import { Pause, Play, Wrench } from 'lucide-react';
+import { toast } from '@/hooks/use-toast';
+
+interface TimerProps {
+    isHost: boolean;
+    startingTime: number;
+    isActive?: boolean;
+}
+
+const Timer: React.FC<TimerProps> = ({ isHost = false, startingTime, isActive = false }) => {
+    const hasInitialized = useRef(false);
+    const [time, setTime] = useState(startingTime);
+    const [timerIsActive, setTimerIsActive] = useState(isActive);
+    const [intervalId, setIntervalId] = useState<NodeJS.Timeout | null>(null);
+
+    const formatTime = (secondsRemaining: number) => {
+        const hours = Math.floor(secondsRemaining / 3600);
+        const minutes = Math.floor((secondsRemaining % 3600) / 60);
+        const seconds = secondsRemaining % 60;
+        return `${hours.toString().padStart(2, '0')}:${minutes.toString().padStart(2, '0')}:${seconds.toString().padStart(2, '0')}`;
+    };
+
+    useEffect(() => {
+        if (isActive && !hasInitialized.current) {
+            hasInitialized.current = true;
+            handleStartTimer();
+        }
+    });
```

Comment:
This `hasInitialized` business isn't necessary if you do
```suggestion
    useEffect(() => {
        if (isActive) {
            handleStartTimer();
        }
    }, [isActive]);
```